BetSystem/BusinnessLogic/SportEventService.cs

```diff
@@ -62,5 +67,98 @@ public void DeleteById(int id)
+            _dbContext.SaveChanges();
+        }
+
+        public List<TeamDto> GetTeamsFromSportEvent(int id)
+        {
+            return _dbContext.Events
+                .Include(x => x.Teams)
+                .FirstOrDefault(x => x.Id == id)
+                .Teams
+                .Select(x => new TeamDto
+                {
+                    Id = x.Id,
+                    Name = x.Name,
+                }).ToList();
+        }
+
+        public void AddBetToSportEvent(BetOnEventDto betOnEventDto)
+        {
+            var betEventToSave = new BetOnEvent
+            {
+                BetOnResult = betOnEventDto.BetOnResult,
+                Amount = betOnEventDto.Amount,
+                Interest = betOnEventDto.Interest,
+                AmountToPay = betOnEventDto.AmountToPay,
+                Team = _dbContext.Teams.First(x => x.Id == betOnEventDto.TeamId),
+                Event = _dbContext.Events.First(x => x.Id == betOnEventDto.EventId)
+            };
+            _dbContext.Bets.Add(betEventToSave);
+            _dbContext.SaveChanges();
+        }
+
+        public List<BetOnEventDto> GetBetsFromSportEvent(int id)
+        {
+            return _dbContext.Events
+                .Include(x => x.Bets)
+                .FirstOrDefault(x => x.Id == id)
+                .Bets
+                .Select(x => new BetOnEventDto
+                {
+                    Id = x.Id,
+                    BetOnResult = x.BetOnResult,
+                    Amount = x.Amount,
+                    Interest = x.Interest,
+                    AmountToPay = x.AmountToPay,
+                }).ToList();
+        }
+
+        public void AddResultToSportEvent(int id, EventResultDto eventResultDto)
+        {
+            var eventResult = new EventResult
+            {
+                BetOnResult = eventResultDto.BetOnResult,
+                Team = _dbContext.Teams.First(x => x.Id == eventResultDto.TeamId),
+                Event = _dbContext.Events.First(x => x.Id == id)
+            };
+            _dbContext.Bets
+                .Include(x => x.Team)
+                .Where(x => x.Event.Id == id)
+                .ToList()
+                .ForEach(x => x.UpdateAmountToPay(eventResult));
+            _dbContext.Results.Add(eventResult);
+            _dbContext.SaveChanges();
+        }
+
+        public List<EventResultDto> GetResultsFromSportEvent(int id)
+        {
+            return _dbContext.Events
+                .Include(x => x.EventResults)
+                .FirstOrDefault(x => x.Id == id)
+                .EventResults
+                .Select(x => new EventResultDto
+                {
+                    Id = x.Id,
+                    BetOnResult = x.BetOnResult,
+                })
+                .ToList();
+            
+        }
+
+        public void PutSportEvent(int id, SportEventDto sportEventDto)
+        {
+            var sportEvent = _dbContext.Events.FirstOrDefault(x => x.Id == id);
```

Comment:
Can we select just name not the whole object?

Reply:
I need whole object to update the Name not return the Name